cdap-api/src/main/java/co/cask/cdap/api/lineage/field/ReadOperation.java

```diff
@@ -68,4 +69,25 @@ public EndPoint getSource() {
   public List<String> getOutputs() {
     return outputs;
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
```

Comment:
these first two if's are already taken care of by super.equals()

Reply:
revised. The first one should be checked in each subclass, but is not needed in Operation because that is abstract.

Reply:
fixed.